kv: TestStoreRangeMergeConcurrentRequests failed due to "SuggestedRange not different from original desc"

https://teamcity.cockroachdb.com/viewLog.html?buildId=2050046&buildTypeId=Cockroach_UnitTests
```
[01:44:41]i:		 [github.com/cockroachdb/cockroach/pkg/kv/kvserver] TestStoreRangeMergeConcurrentRequests
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] gotest://github.com/cockroachdb/cockroach/pkg/kv/kvserver#TestStoreRangeMergeConcurrentRequests
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.623128973-04:00","Action":"run","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
=== RUN   TestStoreRangeMergeConcurrentRequests
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.62314325-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"=== RUN   TestStoreRangeMergeConcurrentRequests\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.644034441-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"W200630 01:44:39.643922 95351 kv/kvserver/store.go:1656  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.645758944-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.645623 95137 gossip/gossip.go:403  [n1] NodeDescriptor set to node_id:1 address:\u003cnetwork_field:\"tcp\" address_field:\"127.0.0.1:40011\" \u003e attrs:\u003c\u003e locality:\u003c\u003e ServerVersion:\u003cmajor_val:0 minor_val:0 patch:0 unstable:0 \u003e build_tag:\"\" started_at:0 cluster_name:\"\" sql_address:\u003cnetwork_field:\"\" address_field:\"\" \u003e \n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.707908504-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.707809 95292 kv/kvserver/queue.go:582  [s1,r3/1:/System/{NodeLive…-tsd}] rate limited in MaybeAdd (merge): throttled on async limiting semaphore\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.708520347-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.708414 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r35] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.723219147-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.720556 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r35:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=1] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.726375538-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.726273 95302 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=3972ed86] removing replica r35/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.72660622-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.726522 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r36] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.829319237-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.829171 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r36:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=3] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.834793765-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.834714 95320 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=f6264561] removing replica r36/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.835418073-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.835371 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r37] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.871639454-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"W200630 01:44:39.871516 96424 kv/kvserver/intentresolver/intent_resolver.go:440  failed to push during intent resolution: failed to push meta={id=1984b231 key=/Local/Range/System/\"tse\"/RangeDescriptor pri=0.00031679 epo=0 ts=0.000000123,11914 min=0.000000123,11914 seq=3} lock=true stat=PENDING rts=0,0 wto=false max=0,0\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.901023997-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.900883 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r37:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=5] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.911473079-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.911371 95336 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=475bd3fc] removing replica r37/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.911890026-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.911832 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r38] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.9814156-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.981247 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r38:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=7] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.989187994-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.989085 95294 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=31a90815] removing replica r38/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:39.989604535-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:39.989532 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r39] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.085836762-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.085685 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r39:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=9] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.093908813-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.093815 95304 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=9bccb53e] removing replica r39/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.094365904-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.094137 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r40] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.167289576-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.167156 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r40:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=11] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.190598071-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.190459 95323 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=8ee7b7d5] removing replica r40/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.192666784-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.192562 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r41] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.308550972-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.308392 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r41:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=13] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.438693685-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.438586 95342 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=903632e1] removing replica r41/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.439743437-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.439684 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r42] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.535174429-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.535042 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r42:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=15] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.551713455-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.551632 95300 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=cc8934e0] removing replica r42/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.552090308-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.552027 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r43] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.655556944-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.655378 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r43:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=17] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.688867721-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.688596 95328 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=a1ece156] removing replica r43/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.691065375-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.690991 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r44] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.721519419-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.721360 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r44:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=19] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.724662534-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.724562 95330 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=e8ead0b5] removing replica r44/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.724909132-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.724799 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r45] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.788782423-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.788637 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r45:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=21] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.796967329-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.796822 95345 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=98acf20b] removing replica r45/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.797956329-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.797865 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r46] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.884958009-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.884823 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r46:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=23] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.904655857-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.904493 95297 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=0cd09c50] removing replica r46/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.905490529-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.905313 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r47] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.97811192-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.976904 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r47:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=25] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.984685215-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.984600 95313 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=d41f69b8] removing replica r47/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:40.985185899-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:40.985124 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r48] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.108554789-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.108410 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r48:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=27] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.113645601-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.113566 95334 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=83025a60] removing replica r48/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.113974136-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.113847 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r49] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.220347006-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.220121 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r49:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=29] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.224845724-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.224706 95286 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=834ba893] removing replica r49/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.225507027-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.225409 95137 kv/kvserver/replica_command.go:413  [s1,r5/1:/{Systemtse-Table/System…}] initiating a split of this range at key \"b\" [r50] (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.346806469-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.346614 95137 kv/kvserver/replica_command.go:670  [s1,r5/1:{/Systemtse-b}] initiating a merge of r50:{b-/Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=31] into this range (manual)\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.366052885-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.365932 95310 kv/kvserver/store_remove_replica.go:122  [s1,r5/1:{/Systemtse-b},txn=abb8c596] removing replica r50/1\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
    TestStoreRangeMergeConcurrentRequests: client_merge_test.go:1621: SuggestedRange not different from original desc. desc: r5:/{Systemtse-Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=24]. suggested: r5:/{Systemtse-Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=26]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.366630266-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"    TestStoreRangeMergeConcurrentRequests: client_merge_test.go:1621: SuggestedRange not different from original desc. desc: r5:/{Systemtse-Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=24]. suggested: r5:/{Systemtse-Table/SystemConfigSpan/Start} [(n1,s1):1, next=2, gen=26]\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.366850763-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.366652 100934 util/stop/stopper.go:539  quiescing\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.367369363-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.367299 100933 util/stop/stopper.go:539  quiescing\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.370014608-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"I200630 01:44:41.367629 100933 util/stop/stopper.go:539  quiescing\n"}
[01:44:41]i:			 [TestStoreRangeMergeConcurrentRequests] [Test Output]
--- FAIL: TestStoreRangeMergeConcurrentRequests (1.75s)
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.37003194-04:00","Action":"output","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Output":"--- FAIL: TestStoreRangeMergeConcurrentRequests (1.75s)\n"}
[01:44:41]i:			 [run] {"Time":"2020-06-29T21:44:41.370040546-04:00","Action":"fail","Package":"github.com/cockroachdb/cockroach/pkg/kv/kvserver","Test":"TestStoreRangeMergeConcurrentRequests","Elapsed":1.75}
```

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -44,7 +44,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -1635,9 +1634,6 @@ func TestStoreRangeMergeCheckConsistencyAfterSubsumption(t *testing.T) {
 func TestStoreRangeMergeConcurrentRequests(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	// Skipping as part of test-infra-team flaky test cleanup.
-	skip.WithIssue(t, 50795)
 
 	ctx := context.Background()
 	storeCfg := kvserver.TestStoreConfig(nil)

